docsk8s/setting-up/ls-k8s-secure.asciidoc

```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
+It's important to set up transport layer security (TLS) to ensure safe communication between the Elastic Stack components running in {k8s}.
+Let's take {filebeat} and {ls} TLS mutual verification as an example. {ls} serves as the server side, while {filebeat} is the client.
 
-[[security-eck-secrets]]
+Create a `Secret` containing server and client SSH keys:
 
-==== Using secrets from an ECK installation
-Description.
+[source,sh]
+--
+kubectl create secret generic logstash-beats-tls --from-file=ca.crt --from-file=client.crt --from-file=client.key --from-file=server.crt --from-file=server.pkcs8.key
+--
 
-[[security-logstash-keystore]]
+On {ls}, configurate the server certificates to the pipeline:
+
+[source,ruby]
+--
+input {
+    beats {
+        port => "5044"
+        ssl => true
+        ssl_certificate_authorities => ["/usr/share/logstash/config/ca.crt"]
+        ssl_certificate => "/usr/share/logstash/config/server.crt"
+        ssl_key => "/usr/share/logstash/config/server.pkcs8.key"
+        ssl_verify_mode => "force_peer"
+    }
+}
+--
+
+Mount the keys we just created to {ls} `Deployment`:
+
+[source,yaml]
+--
+          volumeMounts:
+            - name: logstash-beats-tls
+              mountPath: /usr/share/logstash/config/ca.crt
+              subPath: ca.crt
+            - name: logstash-beats-tls
+              mountPath: /usr/share/logstash/config/server.pkcs8.key
+              subPath: server.pkcs8.key
+            - name: logstash-beats-tls
+              mountPath: /usr/share/logstash/config/server.crt
+              subPath: server.crt
+      volumes:
+        - name: logstash-beats-tls
+          secret:
+            secretName: logstash-beats-tls
+--
+
+On {filebeat}, configurate the client certificates:
+
+[source,yaml]
+--
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: demo
+spec:
+  type: filebeat
+  config:
+    output.logstash:
+      ssl.certificate_authorities: ["/usr/share/filebeat/ca.crt"]
+      ssl.certificate: "/usr/share/filebeat/client.crt"
+      ssl.key: "/usr/share/filebeat/client.key"
+(...)
+  deployment:
+    podTemplate:
+      spec:
+        containers:
+          - name: filebeat
+            volumeMounts:
+              - name: logstash-beats-tls
+                mountPath: /usr/share/filebeat/ca.crt
+                subPath: ca.crt
+              - name: logstash-beats-tls
+                mountPath: /usr/share/filebeat/client.key
+                subPath: client.key
+              - name: logstash-beats-tls
+                mountPath: /usr/share/filebeat/client.crt
+                subPath: client.crt
+        volumes:
+          - name: logstash-beats-tls
+            secret:
+              secretName: logstash-beats-tls
+--
+
+[[security-k8s-secret]]
+==== Using secret
+
+One way to config confidential data, such as password, is to use `Secret` and references it as container environment variable.
 
```

Comment:
We should probably add a note here that this is for illustration purposes and, in production, managing k8s secrets should be done using recognized [good practices](https://kubernetes.io/docs/concepts/security/secrets-good-practices/) to ensure the protection of sensitive information.